osu.Game/Graphics/UserInterface/SettingsDropdownContainer.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public abstract class SettingsDropdownContainer : Container
+    {
+        /// <summary>
+        /// The title of this container, which will be written in header.
+        /// </summary>
+        protected abstract string Title { get; }
+
+        private const float transition_duration = 600;
+        private const int container_width = 270;
+        private const int border_thickness = 2;
+        private const int header_height = 30;
+        private const int corner_radius = 5;
+
+        private FillFlowContainer content;
+        private SimpleButton button;
+        private bool buttonIsActive;
+
+        private Color4 buttonActiveColour;
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            AutoSizeAxes = Axes.Y;
+            Width = container_width;
+            Masking = true;
+            CornerRadius = corner_radius;
+            BorderColour = Color4.Black;
+            BorderThickness = border_thickness;
+
+            Children = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = Color4.Black,
+                    Alpha = 0.5f,
+                },
+                new FillFlowContainer
+                {
+                    Direction = FillDirection.Vertical,
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+
+                    Children = new Drawable[]
+                    {
+                        new Container
+                        {
+                            Name = @"Header",
+                            RelativeSizeAxes = Axes.X,
+                            Height = header_height,
+                            Origin = Anchor.TopCentre,
+                            Anchor = Anchor.TopCentre,

```

Comment:
Prefer Origins + Anchors at the top of the initializer for consistency (there are other cases in this PR where this is not the case).